contracts/dapis/DapiServer.sol

```diff
@@ -636,119 +636,117 @@ contract DapiServer is
                 }
                 unchecked {
                     ind++;
                 }
             }
-            require(signatureCount == 0, "Less signatures than stated");
-            bytes32 beaconSetId = deriveBeaconSetId(beaconIds);
-            uint32 updatedTimestamp = uint32(
-                accumulatedTimestamp / beaconCount
+            // "Greater than or equal to" is not enough because full control
+            // of aggregation requires an absolute majority
+            require(
+                validSignatureCount > beaconCount / 2,
+                "Not enough signatures"
             );
             require(
-                updatedTimestamp >
-                    oevProxyToIdToDataFeed[oevProxy][beaconSetId].timestamp,
-                "Does not update timestamp"
+                dataFeedId == deriveBeaconSetId(beaconIds),
+                "Beacon set ID mismatch"
             );
-            int224 updatedValue = int224(median(values));
-            oevProxyToIdToDataFeed[oevProxy][beaconSetId] = DataFeed({
-                value: updatedValue,
-                timestamp: updatedTimestamp
-            });
-            oevProxyToBalance[oevProxy] += msg.value;
             emit UpdatedOevProxyBeaconSetWithSignedData(
-                beaconSetId,
+                dataFeedId,
                 oevProxy,
                 updateId,
                 updatedValue,
                 updatedTimestamp
             );
         } else if (beaconCount == 1) {
-            (
-                bytes32 beaconId,
-                int224 updatedValue,
-                uint32 updatedTimestamp
-            ) = decodeOevSignedData(metadataHash, signedData[0]);
-            require(updatedTimestamp != 0, "Missing data");
-            require(
-                updatedTimestamp >
-                    oevProxyToIdToDataFeed[oevProxy][beaconId].timestamp,
-                "Does not update timestamp"
-            );
-            oevProxyToIdToDataFeed[oevProxy][beaconId] = DataFeed({
-                value: updatedValue,
-                timestamp: updatedTimestamp
-            });
-            oevProxyToBalance[oevProxy] += msg.value;
+            {
+                (
+                    bool signatureIsNotOmitted,
+                    bytes32 beaconId
+                ) = unpackAndValidateOevUpdateSignature(
+                        oevUpdateHash,
+                        packedOevUpdateSignatures[0]
+                    );
+                require(signatureIsNotOmitted, "Missing signature");
+                require(dataFeedId == beaconId, "Beacon ID mismatch");
+            }
             emit UpdatedOevProxyBeaconWithSignedData(
-                beaconId,
+                dataFeedId,
                 oevProxy,
                 updateId,
                 updatedValue,
                 updatedTimestamp
             );
         } else {
-            revert("Specified no Beacons");
+            revert("Did not specify any Beacons");
```

Comment:
WDYT of moving this to a `require` at the beginning of the function to avoid wasting gas on accessing storage, encoding and decoding if `packedOevUpdateSignatures.length == 0`?

Reply:
That would ever so slightly increase the gas cost of the happy paths. People should never call `updateOevProxyDataFeedWithSignedData()` with an empty `packedOevUpdateSignatures` array anyway so I sacrificed it. (I also expect `updateOevProxyDataFeedWithSignedData()` to be used with Beacon sets, which is why the first condition checks if a Beacon set is being updated.)